File: 프로그래머스0(31~60번)/37.길이에 따른 연산.js
// 레벨 0
// https://school.programmers.co.kr/learn/courses/30/lessons/181879

/*
정수가 담긴 리스트 num_list가 주어질 때, 
리스트의 길이가 11 이상이면 리스트에 있는 모든 원소의 합을 10 이하이면 
모든 원소의 곱을 return하도록 solution 함수를 완성해주세요.
*/

function solution(num_list) {
    var answer = 0;
    
    if (num_list.length >= 11) {
        answer = num_list.reduce((acc, cur) => acc + cur, 0); // 초기누적값 0
    } else {
        answer = num_list.reduce((acc, cur) => acc * cur, 1); // 곱셈이라 최소 1
    }
    
    return answer;
}

File: 프로그래머스 코테(java)/프로그래머스0(31~60번)/37.길이에 따른 연산.java
// 레벨 0
// https://school.programmers.co.kr/learn/courses/30/lessons/181879

class Solution {
    public int solution(int[] num_list) {
        int answer = 0;

        if (num_list.length >= 11) {
            for (int i = 0; i < num_list.length; i++) {
                answer += num_list[i];
            }
        } else {
            answer = 1;
            for (int i = 0; i < num_list.length; i++) {
                answer *= num_list[i];
            }
        }
        return answer;
    }
}